src/common/opencl.c

```diff
@@ -1833,7 +1838,7 @@ int dt_opencl_build_program(const int dev, const int prog, const char *binname,
           //store has using a simple filerename
           char finalfilename[PATH_MAX] = { 0 };
           snprintf(finalfilename, sizeof(finalfilename), "%s" G_DIR_SEPARATOR_S "%s.%s", cachedir, bname, md5sum);
-          rename(link_dest, finalfilename);
+          if(g_rename(link_dest, finalfilename) != 0) goto ret;
```

Comment:
Yeah, this doesn't help, and actually hurts because of chdir. chdir actually should be removed from this code, it doesn't make sense.

Reply:
I'm not sure I can follow you. What is the issue with the chdir? That was in the original code, which I have not touched. Could you pls elaborate a bit?

Reply:
Original code is bad - it shouldn't do chdir. By jumping out with "goto ret" second chdir (restoring original working directory) is skipped, so this code can end up changing the working directory. That's why I'm saying you made it worse. Doing nothing actually was better in this case. If you want to handle rename error, you need to think how to do it, not just copy code from neighbourhood - there's no guarantee that original code makes sense. E.g. you can at least print some error message. Maybe you could try to remove the finalfilename if it exists, though it makes sense to do before calling rename, not after it fails.
To summarize - think about what you are trying to achieve here. Checking return code if you don't know what to do with it doesn't help anyone, it looks like this code handles a error condition, when in reality it doesn't.

Reply:
Fair comment. Let's then just focus on the issue at hand and leave fixing the potential rename error together with the fixing of the original code later.

Reply:
Yes, just leave it as it is for now.